### ISSUE TITLE
Projects created with V1 endpoint ->UI not filtering the namespaces on local cluster.

### ISSUE THREAD

**Issue by dasarinaidu** [rancher/dashboard#9880]
When I create a project on a downstream cluster with kubectl/V1 endpoint -> The project is created and displaying as a project in Downstream cluster but it is also displaying the namespace in Local cluster.

**Steps:**

1. Provision rancher v2.8-head
2. Create downstream cluster
3. Create a project from kubectl/V1 endpoint
4. Copy the local cluster KUBECONFIG and export it on to terminal.
5.  Export the kubeconfig and run on local cluster
6.  `kubectl apply -f ~/Downloads/create_project.yaml`

```
apiVersion: management.cattle.io/v3
kind: Project
metadata:
  name: dastest1
  namespace: c-4k7jp
spec:
  clusterName: c-4k7jp

```
7. Check for the Project display on local and downstream clusters

**Actual:**

a. Project created successfully 
b. On Downstream cluster -> Projects/Namespaces -> The project is displayed (This is expected)
c. On Local cluster -> Projects/Namespaces -> The project is displayed under ->Not in a Project(This is not expected)

**Expected:**
a. UI should filter these namespaces in Local cluster and those should not display

**Note:**
When we create a project via V3 the projects are created and displayed in only at Downstream cluster only NOT displaying in Local cluster. 

![image](https://github.com/rancher/dashboard/assets/22034884/cc24c332-627d-4fd6-9bd9-5221139daa09)

**Comment by mantis-toboggan-md** [comment]
@dasarinaidu under user prefs (`<rancher url>/dashboard/prefs`) do you have "Show system Namespaces managed by Rancher" enabled?

**Comment by dasarinaidu** [comment]
@mantis-toboggan-md I did not enabled anything on the settings just using the default settings only. i.e the field "Show system Namespaces managed by Rancher" is unchecked.


![image](https://github.com/rancher/dashboard/assets/22034884/a9aba0a5-2114-4f30-b84a-16a83e646596)

Note:
I did little more triage and seeing the namespaces built on V3 are generated by system and the namespaces started with p-xxxxxx. Where as in V1 the namespaces (name) is provided by user. Those are not started with p-something. Looks like UI is filtering based the namespace started with p-xxxxx.

**Comment by aalves08** [comment]
Several questions @dasarinaidu : 
- Where was this kubectl applied? In local cluster? In the downstream cluster?
- What is the `c-4k7jp` clusterName? Is this v1 naming for the downstream cluster? v3?

Thanks in advance

**Comment by gaktive** [comment]
@mantis-toboggan-md was able to repro this and this isn't working as expected. It may be a change in the V1 endpoint that requires a change in the namespace filter but this could be a difference in the Norman vs. Steve APIs.

**Comment by mantis-toboggan-md** [comment]
I was able to repro this. I used kubectl to create a project in a downstream cluster, using the mgmt cluster name for `spec.clusterName. A namespace was created in the local cluster with the same name as the project. 

When projects are created through norman (which is what the ui currently does), the namespace created in the local cluster has a name that starts with `p-` ... this is how we're identifying namespaces in the local cluster as projects in downstream clusters. We need to investigate if there's another way of determining that a ns in the local cluster maps to a project downstream, or if we need backend changes to support this.

**Comment by torchiaf** [comment]
@dasarinaidu  I was able to replicate the issue (thanks @mantis-toboggan-md ) and I tried to filter out the downstream projects from local's namespaces list, but I found a back-end inconsistency when new projects are created using `v1` API, which will block also a correct handling in the UI.

- Even if we can filter namespaces, with the same name of a project in another cluster, this will not work when a namespace `dastest3` is defined in `local` cluster and another user creates a project with name 'dastest3' in the `downstream` . `dastest3` namespace wouldn't be displayed in local cluster's namespace list.
- In case of `dastest3` project is created in the downstream, it's not possible to create a new namespace in local cluster, with the same name, the backend throws an error since the namespace already exists:
    
    
    ![Screenshot from 2024-01-30 15-26-31](https://github.com/rancher/dashboard/assets/26394656/9c8b553f-1319-4164-88c5-c2f233c1c74c)
    
    ![Screenshot from 2024-01-30 15-26-01](https://github.com/rancher/dashboard/assets/26394656/8a37c8b7-eeac-42d0-ab51-8708bcb1c8e9)

- In case of a `dastest3` namespace exists in `local` cluster and the user creates a new project `dastest3` in the downstream, when the `dastest3` project is deleted, then the backend deletes also the namespace in `local` cluster, which was existing before the test case.

**Comment by gaktive** [comment]
@torchiaf do we need to file a new ticket in `rancher/rancher` to unblock us? or are you suggesting we transfer this ticket over?

**Comment by torchiaf** [comment]
Blocker issue: https://github.com/rancher/rancher/issues/44280

**Comment by torchiaf** [comment]
@dasarinaidu we got fedback from `rancher` back-end team and both the UI and rancher are working as intended in this case. The correct way to create new projects for downstream clusters using v1 API is to fill the YAML file as follows:

```
apiVersion: management.cattle.io/v3
kind: Project
metadata:
  generateName: p-
  namespace: c-4k7jp
spec:
  clusterName: c-4k7jp
  displayName: dastest1
```
https://github.com/rancher/rancher/issues/44280#issuecomment-1924058380

**Comment by richard-cox** [comment]
I would agree with Michael, this mechanism is working as intended. The UI is also doing the correct thing, given that mechanism.

System namespaces (included namespaces which are downstream projects) are hidden by default. They're [counted as system](https://github.com/rancher/dashboard/blob/master/shell/models/namespace.js#L81) if
- the annotation `management.cattle.io/system-namespace` is set to true
- name is in a hardcoded list of names 
- the name ends in `-system` (although there's an issue somewhere to remove this)
- it's in a project which is classed as system

So creating in the [format](https://github.com/rancher/dashboard/issues/9880#issuecomment-1926673560) above should result in the namespace being hidden by default, visible when the pref is changed as in another comment above. 

I'm assuming there's other functionality that breaks if downstream projects are created outside of that format, so we can't depart from it.

As we move to support more actions via API, in particular steve and not norman, this needs to be clearly documented somewhere. Either in docs site or the `explain` for projects. @MbolotSuse do you know if this is the case?

**Comment by richard-cox** [comment]
I've looked at this some more, playing catch up

The original yaml is valid

```
apiVersion: management.cattle.io/v3
kind: Project
metadata:
  name: dastest1
  namespace: c-4k7jp
spec:
  clusterName: c-4k7jp
```

However we need to make it clear in docs that when generating projects we recommend using `metadata.generateName: p-` and `spec.displayName: <name of your choice>`.

- Supplying metadata generateName / name or spec.displayName are optional (won't break rancher).
- When using `metadata.generateName: p-` rancher by default will filter entry from upstream project/namespace list. This is dependent on the per user `Show system Namespaces managed by Rancher (not intended for editing or deletion)` preference.
- When not using `metadata.generateName: p-` rancher by default will NOT filter entry from upstream project/namespace list
- In all cases upstream namespace related to project has the `management.cattle.io/system-namespace": "true"` namespace, which means it's taken into account for the namespace header filter `Only User Namespaces` / `Only System Namespaces` options

The concept of system filters confused me, as we have global p- ones, and the namespace filter ones.

So going forward this is still a docs change

**Comment by richard-cox** [comment]
I think this is reasonably covered by https://ranchermanager.docs.rancher.com/api/workflows/projects#creating-a-basic-project. Using that example will result in the correct filter behaviour (upstream namespace filtered, given pref setting)
